crates/sui-graphql-rpc/src/types/system_state_summary.rs

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use super::{
+    big_int::BigInt, gas::GasCostSummary, safe_mode::SafeMode, stake_subsidy::StakeSubsidy,
+    storage_fund::StorageFund, system_parameters::SystemParameters,
+};
+use async_graphql::*;
+use sui_types::sui_system_state::sui_system_state_summary::SuiSystemStateSummary as NativeSystemStateSummary;
+
+#[derive(Clone, Debug)]
+pub(crate) struct SystemStateSummary {
+    pub native: NativeSystemStateSummary,
+}
+
+/// Aspects that affect the running of the system that are managed by the validators either
+/// directly, or through system transactions.
+#[Object]
+impl SystemStateSummary {
+    /// SUI set aside to account for objects stored on-chain, at the start of the epoch.
+    /// This is also used for object rebates.
+    async fn storage_fund(&self) -> Option<StorageFund> {
+        Some(StorageFund {
+            total_object_storage_rebates: Some(BigInt::from(
+                self.native.storage_fund_total_object_storage_rebates,
+            )),
+            non_refundable_balance: Some(BigInt::from(
+                self.native.storage_fund_non_refundable_balance,
+            )),
+        })
+    }
+
+    /// Information about whether this epoch was started in safe mode, which happens if the full epoch
+    /// change logic fails for some reason.
+    async fn safe_mode(&self) -> Option<SafeMode> {
+        Some(SafeMode {
+            enabled: Some(self.native.safe_mode),
+            gas_summary: Some(GasCostSummary {
+                computation_cost: self.native.safe_mode_computation_rewards,
+                storage_cost: self.native.safe_mode_storage_rewards,
+                storage_rebate: self.native.safe_mode_storage_rebates,
+                non_refundable_storage_fee: self.native.safe_mode_non_refundable_storage_fee,
+            }),
+        })
+    }
+
+    /// The value of the `version` field of `0x5`, the `0x3::sui::SuiSystemState` object.  This
+    /// version changes whenever the fields contained in the system state object (held in a dynamic
+    /// field attached to `0x5`) change.
+    async fn system_state_version(&self) -> Option<BigInt> {
```

Comment:
As per an earlier comment, let's just make this an `Int` (just return the `u64`) because this is also a number that will not grow quickly enough to hit 2^53.